packages/light.js/src/rpc/other/makeContract.ts

```diff
@@ -32,7 +32,10 @@ interface MakeContract {
  * @return - The contract object as defined in @parity/api.
  */
 const getContract = memoizee(
-  (address: Address, abiJson: any[]) => api().newContract(abiJson, address), // use types from @parity/abi
+  (address: Address, abiJson: any[], provider: any) => {
+    const api = provider ? createApiFromProvider(provider) : getApi();
+    return api.newContract(abiJson, address);
+  }, // use types from @parity/abi
   { length: 1 } // Only memoize by address
```

Comment:
memoization needs to be done based on address & `api` (`provider ? createApiFromProvider(provider) : getApi()`), otherwise if we do `setApi() getContract() setApi() getContract()` the last `getContract()` will return the memoized return value from the first `getContract()` call with the old api